Sofa/Component/LinearSolver/Direct/tests/SparseLDLSolver_test.cpp

```diff
@@ -178,3 +181,50 @@ TEST(SparseLDLSolver, MatrixFactorization)
     static const sofa::type::vector<int> expected_LT_colptr_Values { 0, 0, 0, 0, 0, 0, 1, 3, 3, 3, 3, 4, 4, 5, 7, 11 };
     EXPECT_EQ(invertData->LT_colptr, expected_LT_colptr_Values);
 }
+
+TEST(SparseLDLSolver, EmptySystem)
+{
+    // required to be able to use EXPECT_MSG_NOEMIT and EXPECT_MSG_EMIT
+    sofa::helper::logging::MessageDispatcher::addHandler(sofa::testing::MainGtestMessageHandler::getInstance() ) ;
+
+    using MatrixType = sofa::linearalgebra::CompressedRowSparseMatrix<SReal>;
+
+    MatrixType matrix;
+    matrix.resize(0, 0);
+    matrix.compress();
+
+    using Solver = sofa::component::linearsolver::direct::SparseLDLSolver<MatrixType, sofa::linearalgebra::FullVector<SReal> >;
+    Solver::SPtr solver = sofa::core::objectmodel::New<Solver>();
+
+    solver->init();
```

Comment:
don't you think the init method should also send a warning? no mechanicalObject, empty sytem detected?

Reply:
Fair question. This check is performed at each time step. It visits the mechanical objects and compute the size of the system. I don't know if this can be done in the init method if the mechanical objects are not initialized yet.
Also, the API allows to use a Linear Solver on a matrix, without defining a scene (see the test `TEST(SparseLDLSolver, MatrixFactorization)`). Adding this check in the init method would not make sense in this case.
To be discussed

Reply:
ok I move that discussion in #3504 to not block this PR.